### PR TITLE
Doc/feb16update

### PR DESCRIPTION
**Pull request by TanyaCouture**
Updates and expands index.rst, the Getting Started guide for Mantl. At this time it does not address the possibility of streamlining deployments, which may soon be a Mantl feature.

### CODE REVIEW

File: docs/getting_started/index.rst [hunk 1]
```diff
@@ -1,115 +1,143 @@
...
+
+The Mantl project uses Ansible to bring up
 nodes and clusters. This generally means that you need three things:
 
- 1. hosts to use as the base for your cluster
- 2. an `inventory file`_ with the hosts you want to be modified.
- 3. a playbook to show which components should go where.
+1. hosts to use as the base for your cluster
+2. an `inventory file`_ with the hosts you want to be modified. Mantl includes ansible inventory within its `Dynamic inventory for Terraform.py`_.
+3. a playbook to show which components should go where. Mantl organizes its components in `sample.yml`_, which we recommend copying to ``mantl.yml`` for the possibility of later customization. You can read more about `playbooks`_ in the Ansible docs.
```
Comment (siddharthist) [inline docs/getting_started/index.rst]: Can we get a link to the ansible docs here?

File: docs/getting_started/index.rst [hunk 2]
```diff
@@ -1,115 +1,143 @@
...
+-------------------
+
+The first step for provisioning with any platform is `generating ssh-keys`_ and `secure copying`_ both the public and private keys to your host.
+
+   .. code-block:: shell
 
-.. toctree::
-   :maxdepth: 1
+        ssh-keygen -t rsa -f /path/to/project/sshkey -C "sshkey"
+        scp -P port /path/to/project/id_rsa* <user>@<host>:.ssh/
```
Comment (siddharthist) [inline docs/getting_started/index.rst]: I've actually never had to do this. Maybe note that Terraform does this automatically?

File: docs/getting_started/index.rst [hunk 3]
```diff
@@ -1,115 +1,143 @@
...
 
-You can set up your DNS records with Terraform:
+    .. note:: Greater than one .tf file in existance in the mantl directory will lead to errors upon deployment. If you work with more than one provider, extra .tf files will need to be renamed or moved.
 
-.. toctree::
-   :maxdepth: 2
+Step 3: Run security-setup
+--------------------------
+
+You'll want set up authentication and authorization by running the ``security-setup`` script in the root directory. This will create and set passwords, authentication, and certificates. For more information, see the `security-setup`_ documentation.
```
Comment (siddharthist) [inline docs/getting_started/index.rst]: Can we get a link to that documentation here?

File: docs/getting_started/index.rst [hunk 4]
```diff
@@ -1,115 +1,143 @@
...
 
-   dns.rst
+Step 4: Set up DNS records
+--------------------------
 
+You can set up your DNS records with Terraform: `dns.rst`_
```
Comment (siddharthist) [inline docs/getting_started/index.rst]: Same question here

File: docs/getting_started/index.rst [hunk 5]
```diff
@@ -1,115 +1,143 @@
...
+For the next steps, you may want to add your password to the ssh-agent to maintain a continuous login with your host. One way to do this:
 
-If any servers fail to connect, check your connection by adding ``-vvvv``
-for verbose SSH debugging and try again to view the errors in more detail.
+    .. code-block:: shell
 
-.. warning::
+        eval $(ssh-agent) && ssh-add ~/.ssh/id_rsa
 
-   Due to updated packages in the recent CentOS-7 (1511) release, it is critical
```
Comment (siddharthist) [inline docs/getting_started/index.rst]: Why are we losing the updated packages warning?

File: docs/getting_started/index.rst [hunk 6]
```diff
@@ -1,115 +1,143 @@
...
+Step 4: Deploy the software
+---------------------------
 
-  ansible-playbook -e @security.yml terraform.yml
+   First, you will need to customize a playbook. A sample can be found at ``sample.yml`` in the root directory which you can copy to ``mantl.yml``. You can find more about customizing this at `playbooks`_. The main change you'll want to make is changing ``consul_acl_datacenter`` to your preferred ACL datacenter. If you only have one datacenter, you can remove this variable. Next, assuming you've placed the filled-out template at ``mantl.yml``:
 
-The deployment will probably take a while as all tasks are completed.
+    .. code-block:: shell
+
+        ansible-playbook -e @security.yml mantl.yml
```
Comment (siddharthist) [inline docs/getting_started/index.rst]: This should probably include an `-i` argument

**Comment by siddharthist** [comment]
This will need to be rebased onto master.

File: docs/getting_started/index.rst [hunk 7]
```diff
@@ -1,115 +1,164 @@
 Getting Started
 ===============
 
-.. note:: This document assumes you have a `working Ansible
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: this is actually a RST directive, it needs to be lowercase

File: docs/getting_started/index.rst [hunk 8]
```diff
@@ -1,115 +1,164 @@
...
+.. Note:: This document assumes you have a `working Ansible
           installation`_. If you don't, install Ansible before
           continuing. This can be done simply by running ``pip install -r
           requirements.txt`` from the root of the project.
 
-The Mantl project uses Ansible to bring up nodes and clusters. This generally
-means that you need three things:
+          It also assumes you have a working Terraform installation which you
+          can download from `Terraform downloads`_.
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: This is awkwardly joined - how about "…working Terraform installation. You can download Terraform from…"

File: docs/getting_started/index.rst [hunk 9]
```diff
@@ -1,115 +1,164 @@
...
+1. Hosts to use as the base for your cluster
+2. An `inventory file`_ with the hosts you want to be modified. Mantl includes
+   an Ansible inventory for terraformed hosts, `terraform.py`_.
+3. A playbook matching Ansible roles to hosts. Mantl organizes its components
+   in `sample.yml`_, which we recommend copying to ``mantl.yml`` for the
+   possibility of later customization. You can read more about `playbooks`_ in
+   the `Ansible docs`_.
+
+Preparing to provision Cloud Hosts
+>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: merge marker alert!

File: docs/getting_started/index.rst [hunk 10]
```diff
@@ -1,115 +1,164 @@
...
 (or metal) you care to spin up, as long as it can run CentOS 7 or
-equivalent. However, here are some guides to get started on common
-platforms:
+equivalent.
 
-.. toctree::
-   :maxdepth: 1
+Your hosts will have to be accessible with your SSH key. If you're unfamiliar
+with SSH keys, please read `this blog post
+<https://www.digitalocean.com/community/tutorials/how-to-set-up-ssh-keys--2>`_.
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: ix-nay on the inks-lay that say "this `x`". Try "DigitalOcean's guide to setting up SSH keys" instead. That way everyone knows what they're getting themselves into by clicking the link.

File: docs/getting_started/index.rst [hunk 11]
```diff
@@ -1,115 +1,164 @@
...
+There are several preparatory steps to provisioning the cloud hosts that are
+common to all providers:
 
-Setting up DNS
---------------
+Step 1: Copy .tf file
+---------------------
 
-You can set up your DNS records with Terraform:
+You will also need to copy the .tf file of the platform you are using from
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: "also" in addition to what?

File: docs/getting_started/index.rst (hunk 9 above)
Comment (sehqlr) [inline docs/getting_started/index.rst]: `s/>/^/g`?

File: docs/getting_started/index.rst [hunk 12]
```diff
@@ -1,115 +1,164 @@
...
+need to be changed to your configuration.
 
-.. toctree::
-   :maxdepth: 2
+.. note::
 
-   dns.rst
+    More than one .tf file in the mantl directory will lead to errors upon
+    deployment. If you work with more than one provider, extra .tf files will
+    need to be renamed or moved.
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: First sentence: good. Yes. 2nd sentence needs a little work. Maybe we could just be very specific that we don't do multi-DC that way.

File: docs/getting_started/index.rst [hunk 13]
```diff
@@ -1,115 +1,164 @@
...
 
+Step 2: Run security-setup
+--------------------------
+
+Running the ``security-setup`` script in the root directory will create and set
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: let's remove "create and"

File: docs/getting_started/index.rst [hunk 14]
```diff
@@ -1,115 +1,164 @@
...
+--------------------------
 
-Before you begin, you'll want to run the ``security-setup`` script in the
-root directory. This will create and set passwords, authentication, and
-certificates. For more information, see the :doc:`security-setup
-<../security/security_setup>` documentation.
+You can set up your DNS records with Terraform: `dns.rst`_
+
+Provisioning Cloud Hosts
+>>>>>>>>>>>>>>>>>>>>>>>>
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: Are these merge markers or headers? They looks similar. Elsewhere in the docs, we've used `'''''''''` to indicate third-level headers (although I may be misremembering, check and go with the Sphinx project's common practice if we don't have any)

File: docs/getting_started/index.rst [hunk 15]
```diff
@@ -1,115 +1,164 @@
...
           ``ansible -e ansible_python_interpreter=/path/to/python2``.
 
-In the following examples, we're going to assume you deployed hosts using
-Terraform. This project ships with a dynamic inventory file to read Terraform
-``.tfstate`` files. If you are running ansible from the root directory of the
-project, this inventory file will be used by default. If not, or to use a custom
-inventory file, you can use the ``-i`` argument of ``ansible`` or
-``ansible-playbook`` to specify the inventory file path. For example:
+The following steps assume that you have provisioned your cloud host by taking
+the steps listed in one of the guides listed above. We're going to assume you
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: remove one "listed"

File: docs/getting_started/index.rst [hunk 16]
```diff
@@ -1,115 +1,164 @@
...
+deployed hosts using Terraform.
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: This sentence is now redundant.

File: docs/getting_started/index.rst [hunk 17]
```diff
@@ -1,115 +1,164 @@
...
+
+This project ships with a dynamic inventory file to read Terraform ``.tfstate``
+files, `terraform.py`_.  If you are not running Ansible from the root directory,
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: colon or nothing, not comma

File: docs/getting_started/index.rst [hunk 18]
```diff
@@ -1,115 +1,164 @@
...
+       ansible-playbook -i path/to/inventory -e @security.yml mantl.yml
 
-First, ping the servers to ensure they are reachable via ssh:
+Steps to deploying via Ansible:
+>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+
+Step 1: Add password to the ssh-agent
+-------------------------------------
+
+For the next steps, you may want to `add your password to the ssh-agent
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: good link title :clap:

File: docs/getting_started/index.rst [hunk 19]
```diff
@@ -1,115 +1,164 @@
...
+        eval $(ssh-agent) && ssh-add ~/.ssh/id_rsa
 
-If any servers fail to connect, check your connection by adding ``-vvvv``
-for verbose SSH debugging and try again to view the errors in more detail.
+Step 2: Ping the servers to ensure they are reachable via ssh
+-------------------------------------------------------------
+
+.. code-block:: shell
+
+        ansible all -i plugins/inventory/terraform.py -m ping
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: we can remove `-i plugins/inventory/terraform.py`, it's the default inventory plugin (check `ansible.cfg`)

File: docs/getting_started/index.rst [hunk 20]
```diff
@@ -1,115 +1,164 @@
...
+-------------------------------------
+
+For the next steps, you may want to `add your password to the ssh-agent
+<https://wiki.archlinux.org/index.php?title=SSH_keys&redirect=no#SSH_agents>`_
+avoid re-entering your password.
 
 .. code-block:: shell
 
-  ansible all -m ping
+        eval $(ssh-agent) && ssh-add ~/.ssh/id_rsa
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: is this necessary? The article above surely contains this, and we don't give any other helpful tips for ssh-agent.

File: docs/getting_started/index.rst [hunk 21]
```diff
@@ -1,115 +1,164 @@
...
+Step 2: Ping the servers to ensure they are reachable via ssh
+-------------------------------------------------------------
+
+.. code-block:: shell
+
+        ansible all -i plugins/inventory/terraform.py -m ping
+
+It may take a few minutes after running terraform for the servers to be
+reachable. If any servers fail to connect, you can check your connection by
+adding ``-vvvv`` for verbose SSH debugging and try again to view the errors in
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: remove "and try again"

File: docs/getting_started/index.rst [hunk 22]
```diff
@@ -1,115 +1,164 @@
...
-   Due to updated packages in the recent CentOS-7 (1511) release, it is critical
-   that you upgrade operating system packages on all server before proceeding
-   with deployment:
+        Due to updated packages in the recent CentOS 7 (1511) release, it is
+        critical that you upgrade operating system packages on all servers
+        before proceeding with the deployment:
+
+.. code-block:: shell
+
+        ansible-playbook -e 'serial=0' playbooks/upgrade-packages.yml
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: `serial=0` is the default now, we can remove

File: docs/getting_started/index.rst [hunk 23]
```diff
@@ -131,11 +180,11 @@ If you need the IP address of your nodes, you can use ``terraform.py``:
...
 
 When you enter a control node's IP address into your browser, you'll likely get
-prompted about invalid security certificates if you have SSL/TLS turned on.
-(Follow your browser's instructions on how to access a site without a valid
-cert.) Then, you will be presented with a basic access authentication prompt.
-The username and password for this is based upon the ``security.yml`` file that
-you generated earlier with the ``security-setup`` script.
+prompted about invalid security certificates (if you have SSL/TLS turned on).
+Follow your browser's instructions on how to access a site without a valid
+cert. Then, you will be presented with a basic access authentication prompt.
```
Comment (BrianHicks) [inline docs/getting_started/index.rst]: certs

File: docs/getting_started/index.rst (hunk 9 above)
Comment (BrianHicks) [inline docs/getting_started/index.rst]: Looks like `#`, `*`, `=`, `-`, `^`, `"` officially. So we're already off... But let's maybe use `^`, yes.

**Comment by siddharthist** [comment]
rebased and squashed

**Comment by sehqlr** [comment]
LGTM

**Pull request merged**